src/finch/tensor.py

```diff
@@ -357,12 +358,21 @@ def _from_numpy(cls, arr: np.ndarray, fill_value: np.number) -> JuliaObj:
     def from_scipy_sparse(cls, x) -> "Tensor":
         if not _is_scipy_sparse_obj(x):
             raise ValueError("{x} is not a SciPy sparse object.")
-        if x.format not in ("coo", "csr", "csc"):
-            x = x.asformat("coo")
         return Tensor(x)
 
     @classmethod
     def _from_scipy_sparse(cls, x) -> JuliaObj:
+        if x.format not in ("coo", "csr", "csc"):
+            x = x.asformat("coo")
+        if not x.has_canonical_format:
+            warnings.warn(
+                "SciPy sparse input must be in a canonical format. "
+                "Calling `sum_duplicates`."
+            )
+            x = x.copy()
+            x.sum_duplicates()
+            assert x.has_canonical_format
```

Comment:
Can we have a `copy : bool | None` kwarg that can ensure zero-copy when it needs to be ensured, and pass that in via `asarray`.

Reply:
Can I add `copy` argument in a follow-up PR? I started working on it and it requires a bit more changes in `tensor.py`. For now I would like to be sure that SciPy sparse objects are always loaded correctly. Is it Ok?

Reply:
Sure, I've created #46 to track that.